experimental/dds/tree2/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -68,7 +69,7 @@ const deepSchema = deepBuilder.intoDocumentSchema(
 	SchemaBuilder.field(FieldKinds.value, linkedListSchema, jsonNumber),
 );
 
-const factory = new SharedTreeFactory();
+const factory = new SharedTreeFactory({ validator: typeboxValidator });
```

Comment:
use cases like this will likely gain more configuration options in the future.

We should consider how we want the API to evolve when that happens.

If optional configuration gets added, I'd prefer to add it as members of this configuration object rather than as additional optional parameters (optional parameters have less API compatibility that one might expect).

Therefor I think we might want to do one of:

1. wrap this config in an extra general purpose config object: ex: `new SharedTreeFactory({codec: { validator: typeboxValidator }})`
2. make the members of this config have slightly more specific names so unioning this config object with others is more clear: ex: `new SharedTreeFactory({ jsonValidator: typeboxValidator })`.

At least for required config, the second option seems cleaner.